ui-react/packages/atlasmap/src/Atlasmap/Atlasmap.tsx

```diff
@@ -365,19 +365,29 @@ export const Atlasmap: FunctionComponent<IAtlasmapProps> = ({
     ],
   );
 
+  const handleRemoveMapping = useCallback(
+    () =>
+      handlers.onDeleteMapping(() => {
+        removeMapping(selectedMapping!.mapping);
+        deselectMapping();
+      }),
+    [handlers, deselectMapping, removeMapping, selectedMapping],
+  );
+
   const mappingEvents: IMappingDocumentEvents = useMemo(
     () => ({
       onSelectMapping: selectMapping,
       onDeselectMapping: deselectMapping,
       onEditMapping: () => void 0,
+      onDeleteMapping: handleRemoveMapping,
```

Comment:
This will remove the selected mapping, not the mapping the user asked to be deleted.
Either change the `handleRemoveMapping` to accept the mapping to be deleted as an argument and change the other place where it's used, or create a new handler.